nautobot/utilities/utils.py

```diff
@@ -907,7 +907,9 @@ def get_filterset_parameter_form_field(model, parameter, filterset=None):
 
         form_field = DynamicModelMultipleChoiceField(**form_attr)
     elif isinstance(field, ContentTypeMultipleChoiceFilter):
-        plural_name = model._meta.verbose_name_plural
+        from nautobot.utilities.utils import slugify_dashes_to_underscores
+
+        plural_name = slugify_dashes_to_underscores(model._meta.verbose_name_plural)
```

Comment:
This is still a hack, tbh - we're *assuming* that the slugified model verbose_name_plural matches the feature name in the model features registry... but probably fine for now.

Reply:
Yeah, I commented on the issue about that, and this appears to be the only case where a verbose_name_plural of a model also matches an extras feature, so it's actually specific to dynamic groups... apparently.